keel-sql/src/main/resources/db/changelog/20201116-foreign-keys.yml

```diff
@@ -0,0 +1,213 @@
+        baseTableName: delivery_config_last_checked
+        baseColumnNames: delivery_config_uid
+        constraintName: fk_delivery_config_last_checked_delivery_config
+        referencedTableName: delivery_config
+        referencedColumnNames: uid
+        referencesUniqueColumn: true
+        onDelete: CASCADE
+- changeSet:
+    id: foreign-keys-on-environment
+    author: fletch
+    changes:
+    - addForeignKeyConstraint:
+        baseTableName: environment
+        baseColumnNames: delivery_config_uid
+        constraintName: fk_environment_delivery_config
+        referencedTableName: delivery_config
+        referencedColumnNames: uid
+        referencesUniqueColumn: true
+        onDelete: CASCADE
+- changeSet:
+    id: foreign-keys-on-environment_artifact_constraint
+    author: fletch
+    changes:
+    - sql:
+        sql: delete from environment_artifact_constraint where uid = ''
+    - addForeignKeyConstraint:
+        baseTableName: environment_artifact_constraint
+        baseColumnNames: environment_uid
+        constraintName: fk_environment_artifact_constraint_environment
+        referencedTableName: environment
+        referencedColumnNames: uid
+        referencesUniqueColumn: true
+        onDelete: CASCADE
+- changeSet:
+    id: foreign-keys-on-environment_artifact_pin
+    author: fletch
+    changes:
+    - addForeignKeyConstraint:
+        baseTableName: environment_artifact_pin
+        baseColumnNames: environment_uid
+        constraintName: fk_environment_artifact_pin_environment
+        referencedTableName: environment
+        referencedColumnNames: uid
+        referencesUniqueColumn: true
+        onDelete: CASCADE
+    - addForeignKeyConstraint:
+        baseTableName: environment_artifact_pin
+        baseColumnNames: artifact_uid
+        constraintName: fk_environment_artifact_pin_delivery_artifact
+        referencedTableName: delivery_artifact
+        referencedColumnNames: uid
+        referencesUniqueColumn: true
+        onDelete: CASCADE
+- changeSet:
+    id: foreign-keys-on-environment_artifact_queued_approval
+    author: fletch
+    changes:
+    - sql:
+        sql: delete from environment_artifact_queued_approval where not exists (select 1 from environment where environment.uid = environment_artifact_queued_approval.environment_uid)
+    - addForeignKeyConstraint:
+        baseTableName: environment_artifact_queued_approval
+        baseColumnNames: environment_uid
+        constraintName: fk_environment_artifact_queued_approval_environment
+        referencedTableName: environment
+        referencedColumnNames: uid
+        referencesUniqueColumn: true
+        onDelete: CASCADE
+- changeSet:
+    id: foreign-keys-on-environment_artifact_versions
+    author: fletch
+    changes:
+    - sql:
+        sql: delete from environment_artifact_versions where not exists (select 1 from delivery_artifact where delivery_artifact.uid = environment_artifact_versions.artifact_uid)
```

Comment:
Should we look for orphans based on `environment_uid` as well?

Reply:
I added these delete statements where the FK could not get added on a clone of production data so where there's no delete, there's no orphan data